SimPEG/EM/Static/DC/IODC.py

```diff
@@ -0,0 +1,444 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib
+import properties
+import warnings
+
+import SimPEG
+from SimPEG import Utils, Mesh
+from . import SrcDC as Src   # Pole
+from . import RxDC as Rx
+from .SurveyDC import Survey_ky
+
+
+class IO(properties.HasProperties):
+    """ """
+
+    # Survey
+    survey_geometry = properties.StringChoice(
+        "Survey geometry of DC surveys",
+        default="SURFACE",
+        choices=["SURFACE", "BOREHOLE", "GENERAL"]
+    )
+
+    survey_type = properties.StringChoice(
+        "DC-IP Survey type",
+        default="dipole-dipole",
+        choices=[
+            "dipole-dipole", "pole-dipole",
+            "dipole-pole", "pole-pole"
+        ]
+    )
+
+    dim = properties.Integer(
+        "Dimension of locations",
+        default=2,
+        required=True
+    )
+
+    a_locations = properties.Array(
+        "locations of the positive (+) current electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    b_locations = properties.Array(
+        "locations of the negative (-) current electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    m_locations = properties.Array(
+        "locations of the positive (+) potential electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    n_locations = properties.Array(
+        "locations of the negative (-) potential electrodes",
+        required=True,
+        shape=('*', '*'),  # ('*', 3) for 3D or ('*', 2) for 2D
+        dtype=float  # data are floats
+    )
+
+    electrode_locations = properties.Array(
```

Comment:
is this any different than an hstack of [`a_locations`, `b_locations`, `m_locations`, `m_locations`]? If it is the same, then this should not be a settable property.

Reply:
`electrode_locations ` is different from stack of abmn. it is a unique electrode locations, whereas abmn are locations for all datum.

Reply:
kk, thanks!